Feature/mongodb

rochamarcelo opened the pull request:
Added mongo engine

rochamarcelo committed:
Added mongodb engine

rochamarcelo committed:
Using config 'url' for mongo engine

josegonzalez commented:
Can you update the docs as well?

rochamarcelo commented:
Do you mean this one https://github.com/josegonzalez/php-queuesadilla/blob/master/docs/supported-systems.md?

josegonzalez commented:
That and anything else that might be needed (tests also need to pass, but thats a given).

josegonzalez committed:
chore: add missing newline

josegonzalez commented on tests/josegonzalez/Queuesadilla/Engine/MongoEngineTest.php:
```diff
@@ -0,0 +1,397 @@
+    /**
+     * @inheritDoc
+     */
+    public function setUp(): void
```
This assumes a modern PHP, while the package still supports 5.6.

rochamarcelo replied:
removed.

rochamarcelo committed:
don't use return type syntax

rochamarcelo committed:
don't use return type syntax

rochamarcelo committed:
Merge branch 'feature/mongodb' of github.com:CakeDC/php-queuesadilla into feature/mongodb

rochamarcelo committed:
Added mongodb engine in 'supported systems'

rochamarcelo committed:
mongodb for travis

rochamarcelo committed:
Merge remote-tracking branch 'upstream/master' into feature/mongodb

# Conflicts:
#	.travis.yml

rochamarcelo committed:
fixed return type. now supports php >= 7.2

rochamarcelo committed:
phpcs fix: Avoid short variable name

rochamarcelo commented:
@josegonzalez For some reason travis keeps using the old .travis.yml. We need the env MONGO_URL and mongodb service to run tests

rochamarcelo committed:
mongodb as dev-dependency

rochamarcelo committed:
mongodb as dev-dependency

rochamarcelo commented:
@josegonzalez After some debugging I was able to setup travis to run mongodb tests. It should be good to go now.

josegonzalez commented:
Mind rebasing onto master? Not a merge, a rebase.

rochamarcelo commented:
I don't think I'll have permission to push to master. By the way I have merged master > feature/mongodb

rochamarcelo committed:
Fixed deprecated signature

Fixed ` Return type of AuditStash\Event\BaseEvent::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice `

josegonzalez commented:
If you rebase this with the changes from the current php-queuesadilla, I'll merge it.

rochamarcelo committed:
Use correct key for delay and rename key expiresAt to expires_at which is how it is named in the others engines.

rochamarcelo committed:
Use correct key for delay and rename key expiresAt to expires_at which is how it is named in the others engines.

steinkel committed:
Merge pull request #3 from CakeDC/issue/fix-delay-and-naming

Issue/fix delay and naming
